src/Windowing/Silk.NET.Windowing/Silk.cs

```diff
@@ -14,15 +14,32 @@ namespace Silk.NET.Windowing
     /// </summary>
     public static class Silk
     {
+        private static ISilkPlatform _currentPlatform { get; set; }
+        
         /// <summary>
         /// Gets or sets the current windowing platform.
         /// </summary>
-        public static ISilkPlatform CurrentPlatform { get; set; }
+        /// /// <exception cref="InvalidOperationException">
+        /// Thrown if the set platform is not applicable.
+        /// </exception>
+        public static ISilkPlatform CurrentPlatform
+        {
+            get => _currentPlatform;
+            set
+            {
+                if (!value.IsApplicable) {
+                    throw new InvalidOperationException("Requested platform is not applicable");
```

Comment:
Hmmm, I'm against doing a check in here as power users (who are generally the only people who will ever touch the `Silk` class) might want to override the auto-initialization and use a backend regardless of whether it's applicable or not. If they want to go nuts with their fancy custom platforms, don't let us stop them ;)